docs/sources/flow/reference/components/loki.source.gcplog.md

```diff
@@ -0,0 +1,157 @@
+fields take their default values.
+
+Name                     | Type          | Description | Default | Required
+------------------------ | ------------- | ----------- | ------- | --------
+`project_id`             | `string`      | The GCP project id the subscription belongs to.       |         | yes
+`subscription`           | `string`      | The subscription to pull logs from.                   |         | yes
+`labels`                 | `map(string)` | Additional labels to associate with incoming logs.    | `"{}"`  | no
+`use_incoming_timestamp` | `bool`        | Whether to use the incoming log timestamp.            | `false` | no
+
+To make use of the `pull` strategy, the GCP project must have been
+[configured](https://grafana.com/docs/loki/next/clients/promtail/gcplog-cloud/)
+to forward its cloud resource logs onto a Pub/Sub topic for
+`loki.source.gcplog` to consume.
+
+Typically, the host system also needs to have its GCP
+[credentials](https://cloud.google.com/docs/authentication/application-default-credentials)
+configured. One way to do it is to point the `GOOGLE_APPLICATION_CREDENTIALS`
+environment variable to the location of a credential configuration JSON file or
+a service account key.
+
+### push block
+
+The `push` block defines the configuration of the HTTP server that receives
+push requests from GCP's Pub/Sub servers.
+
+The following arguments can be used to configure the `push` block. Any omitted
+fields take their default values.
+
+Name                     | Type          | Description | Default | Required
+------------------------ | ------------- | ----------- | ------- | --------
+`http_listen_address`    | `string`      | The address the server listens to.  | `"0.0.0.0"` | no
+`http_listen_port`       | `int`         | The port the server listens to.     |  `8080`     | no
+`push_timeout`           | `duration`    | Sets a maximum processing time for each incoming GCP log entry. |  `"0s"`  | no
+`labels`                 | `map(string)` | Additional labels to associate with incoming entries. | `"{}"`  | no
+`use_incoming_timestamp` | `bool`        | Whether to use the incoming entry timestamp.          | `false` | no
+
+The server listens for POST requests from GCP's Push subscriptions on
+`HOST:PORT/gcp/api/v1/push`.
+
+By default, for both strategies the component assigns the log entry timestamp
+as the time it was processed, except if `use_incoming_timestamp` is set to
+true.
+
+The `labels` map is applied to every entry that passes through the component.
+
+
+## Exported fields
+
+`loki.source.gcplog` does not export any fields.
+
+## Component health
+
+`loki.source.gcplog` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`loki.source.gcplog` exposes some debug information per gcplog listener:
+* The configured strategy.
+* Their label set.
+* When using a `push` strategy, the listen address.
```

Comment:
```suggestion
* The configured strategy
* Its label set
* The listen address when using a `push` strategy
```

Reply:
Wasn't sure what their was referring to in "their label set" -- isn't it a single listener?

Reply:
(For these short bullets, where none is a sentence, I removed the periods)